addons/mail/tools/discuss.py

```diff
@@ -157,6 +157,7 @@ def delete(self, records, as_thread=False):
             )
             index = self._get_record_index(model_name, values)
             self._ensure_record_at_index(model_name, index)
+            self._add_values(values, model_name, index)
```

Comment:
Unrelated fix (for earlier versions):

Delete was not working if the record wasn't added in the same request, because it put the "delete" key without any id on the object.